src/storage/core.rs

```diff
@@ -1228,14 +1241,31 @@ where
         }
     }
 
-    async fn fsyncdata(&self) -> IOResult<()> {
-        self.safe.read().await.fsyncdata().await
+    pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
+        let safe = self.safe.read().await;
+        if let Some(ablob) = &safe.active_blob {
+            let ablob = ablob.read().await;
+            if !self.should_try_fsync(ablob.file_dirty_bytes()) {
+                return Ok(());
+            }
+        }
+        self.fsync_in_progress.store(true, Ordering::Release);
```

Comment:
There are 2 problems here:
1. Right now there is no guarantee that `fsync_in_progress` will be reset to `false`, because of `?` on the line below. The best way to solve the problem is to declare a helper struct to use RAII feature. This struct will hold a reference to `AtomicBool` and resets its value to `false` on drop
2. It is better to set `fsync_in_progress` to `true` as early as possible to prevent large number of messages in the `Observer` queue. At the beginning of the function `read` lock is acquired and that can took some time. Thus it is better to set  `fsync_in_progress` to `true` before that (right at the beginning of this function)